pyiron_base/__init__.py

```diff
@@ -46,6 +45,9 @@
 
 from pyiron_base.jobs.job.toolkit import Toolkit, BaseTools
 
+# Expose snippets references in base API for backwards compatibility
+from pyiron_snippets.import_alarm import ImportAlarm
```

Comment:
I find this confusing - should not we do the major change and just enforce the switch to using the `ImportAlarm` only from `pyiron_snippets`?

Reply:
I propose to leave this PR with the slightly-confusing-but-backwards-compatible redirection here, but I stacked an additional, compatibility breaking patch in https://github.com/pyiron/pyiron_base/pull/1458. 

I agree that this might be slightly confusing, but I don't think it's explicitly wrong. So what I'd like to do is get go through and approve each part of the stack individually, then we can merge the whole thing in at once. The mild confusingness will thus still exist at a particular commit, but not be included in any releases.

Reply:
I've converted this PR to a draft until we settle on a solution for the compatibility.